<?xml version="1.0"?>
<launch>
  <!-- Auto-Mapping Launch File -->
  <!-- Tek tÄ±kla: Gazebo + Robot + GMapping + Auto-Explorer -->
  
  <!-- 1. Gazebo World with Robot -->
  <include file="$(find me48b_library_robot)/launch/library_with_robot.launch">
    <arg name="gui" value="true"/>
  </include>
  
  <!-- 1b. Robot State Publisher (Required for TF) -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
    <param name="publish_frequency" type="double" value="50.0"/>
  </node>
  
  <!-- 2. GMapping SLAM -->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
    <param name="base_frame" value="base_footprint"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_frame" value="map"/>
    <param name="map_update_interval" value="2.0"/>
    <param name="maxUrange" value="6.0"/>
    <param name="maxRange" value="8.0"/>
    <param name="sigma" value="0.05"/>
    <param name="kernelSize" value="1"/>
    <param name="lstep" value="0.05"/>
    <param name="astep" value="0.05"/>
    <param name="iterations" value="5"/>
    <param name="lsigma" value="0.075"/>
    <param name="ogain" value="3.0"/>
    <param name="lskip" value="0"/>
    <param name="minimumScore" value="200"/>
    <param name="srr" value="0.01"/>
    <param name="srt" value="0.02"/>
    <param name="str" value="0.01"/>
    <param name="stt" value="0.02"/>
    <param name="linearUpdate" value="0.5"/>
    <param name="angularUpdate" value="0.436"/>
    <param name="temporalUpdate" value="-1.0"/>
    <param name="resampleThreshold" value="0.5"/>
    <param name="particles" value="80"/>
    <param name="xmin" value="-20.0"/>
    <param name="ymin" value="-20.0"/>
    <param name="xmax" value="20.0"/>
    <param name="ymax" value="20.0"/>
    <param name="delta" value="0.05"/>
    <param name="llsamplerange" value="0.01"/>
    <param name="llsamplestep" value="0.01"/>
    <param name="lasamplerange" value="0.005"/>
    <param name="lasamplestep" value="0.005"/>
  </node>
  
  <!-- 3. RViz for Visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find me48b_library_robot)/rviz/mapping.rviz" output="screen"/>
  
  <!-- 4. Auto-Mapper (Delayed Start) -->
  <node pkg="me48b_library_robot" type="auto_mapper.py" name="auto_mapper" output="screen" launch-prefix="bash -c 'sleep 8; $0 $@'"/>
  
</launch>
